libs/clients/recycling-fund/src/lib/recyclingFundClient.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { print } from 'graphql'
+
+import { User } from '@island.is/auth-nest-tools'
+import type { EnhancedFetchAPI } from '@island.is/clients/middlewares'
+import type { ConfigType } from '@island.is/nest/config'
+
+import {
+  CreateRequestDocument,
+  CreateRequestMutation,
+  CreateRequestMutationVariables,
+  SkilavottordRecyclingRequestDocument,
+  SkilavottordRecyclingRequestMutationVariables,
+  SkilavottordVehicleDocument,
+  SkilavottordVehicleMutationVariables,
+  SkilavottordVehicleOwnerDocument,
+  SkilavottordVehicleOwnerMutationVariables,
+} from './createRecyclingRequest.generated'
+
+import { RecyclingRequestTypes } from '../../gen/schema'
+import { RecyclingFundClientConfig } from './recyclingFundClient.config'
+import { RecyclingFundFetchKey } from './recyclingFundFetchProvider'
+
+const baseGqlRequestOptions = {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+  },
+}
+
+@Injectable()
+export class RecyclingFundClientService {
+  constructor(
+    @Inject(RecyclingFundClientConfig.KEY)
+    private readonly config: ConfigType<typeof RecyclingFundClientConfig>,
+    @Inject(RecyclingFundFetchKey)
+    private readonly fetch: EnhancedFetchAPI,
+  ) {}
+
+  async createRecyclingFundRequest(user: User, body: any) {
+    return await this.fetch(this.config.gqlBasePath, {
+      ...baseGqlRequestOptions,
+      auth: user,
+      body: JSON.stringify(body),
+    })
+  }
```

Comment:
The name should be more generic like `gqlRequestWithAuth` and `body` shouldn't be an `any` type. You should be able to type it and use generics to help with typesafety of `query` and `variabes` fields.